### DISCUSSION TITLE
Receiving 'Datasource was invalidated' notification

### DISCUSSION THREAD

**Discussion by alanneese** [dbeaver/dbeaver#15936] [General]
#### System information: 
- Windows 10
- DBeaver 21.3.5.202202200327

#### Connection specification:
- Database name : jhj7bqbh_menudb
- Driver name : MySQL 5
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:
I keep getting the notification from DBeaver 'Datasource was invalidated'  Live connection count: 2/2
#### Steps to reproduce, if exist:
Just access my database and run a query.
#### Include any warning/errors/backtraces from the logs

eclipse.buildId=unknown
java.version=11.0.12
java.vendor=Eclipse Foundation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Command-line arguments:  -os win32 -ws win32 -arch x86_64

org.jkiss.dbeaver.model
Error
Tue Mar 01 15:11:21 CST 2022
Cannot determine transaction isolation level due to connection hanging. Setting to NONE.

**Comment by Matvey16** [comment]
@alanneese Hello, thank you for the report. Do you have any settings that may be disconnecting you because of the timeouts? This message should appear if the connection for editor was lost

**Comment by alanneese** [comment]
No, not that I am aware of.  I have been fighting this error for a while.

**Comment by scottie** [comment]
I get the same issue myself, no different settings all out of the box.

**Reply by JoaoBarreto** [reply]
I'm seeing the same situation with HAProxy + 2 Instances of MariaDB (Galera Cluster). DBeaver connecting to HAProxy IP.

**Comment by ronaldcastro88** [comment]
I have the same problem

**Reply by panipanda** [reply]
Same problem on Kubuntu 22.04, DBeaver 22.1.1. I haven't any specific settings and yet sometimes got this error.

**Reply by mkgrgis** [reply]
This notification usually depends only on network session stability.

**Comment by hoainguyenkatalon** [comment]
I have the same issue in MAC, but is does not in Windows.
In MAC, when I no use the Dbeaver for long time and when I back to use, DBeaver takes long time to reconnect the connection and show the Datasource was invalidated notification.

**Comment by dataguardian** [comment]
I'm seeing the same issue:
Windows 10
Community Edition Version 22.1.4.202208051447

**Reply by Alexhia47** [reply]
same with windows 11

**Comment by dataguardian** [comment]
I am having the same issue.
Installed and have not changed settings.
Version 22.2.0.202209051344 on Windows 10

**Comment by GiovanaBussolotti** [comment]
Someone has solved yet?

**Comment by anubrato** [comment]
I am having same issue. Community edition 22.2.5.202211202223 on MacOS 12.6.1 (Monterey)

**Comment by saRca** [comment]
I am having the same issue. Community edition Versión22.3.0.202212041619 on MacOs 12.5 Monterrey. I see this notice every 60 seconds, when connected to a mysql db on remote DB in a cpanel web host.

**Comment by Alexhia47** [comment]
i cant fix it...windows 11 here

**Comment by CrimsonVex** [comment]
Same issue here, SQLserver

**Comment by MorrisDa** [comment]
same issue

**Comment by KevChaplin** [comment]
Same issue here, Windows 11, connecting to AWS hosted Postgres Db.
Frequent loss of connection with 'Datasource was invalidated' message, often (but not always) slow execution times when working. 
DBeaver Version 22.3.3.202301211053.
Working over wi-fi connection. Previously no issues though.

**Comment by dmarvp** [comment]
I also have the same problem. Using a Mac with an M1 processor.

**Reply by metafaniel** [reply]
I'd really like to migrate from _SQL Developer_ to _DBeaver_ but this is keeping me from using it in my daily work, as I'm unable to even navigate tables content or properties because of this...
I have no particular settings at all, just DBeaver out of the box; I've installed Oracle driver, set my connections properties to establish a connection and that's about it.

If I can help in any way, please let me know.

**Comment by givihuda** [comment]
I've been getting this error randomly over multiple versions of DBeaver and macOS, regardless of connection. Sometimes doing anything with the database auto-reconnects, sometimes I have to right-click the database connection and select Refresh, sometimes I have to restart DBeaver. It's been super-annoying.

**Comment by jacobgawel** [comment]
Im getting the notification, I'm using the most recent version of dbeaver EE:

Version 22.3.0.202212131935

Every time i pull from the DB or interact with the DB it shows the notification. Is there any way to turn it off? Its not really affecting my work but its kinda annoying.

**Comment by PaulDokken** [comment]
Does anybody know why a database server is being invalidated so often.  I am using MySQL 5.7

**Comment by givihuda** [comment]
Most server connections I have in my work, that are not archaic, do not time out easily or quickly. They tend to use keepalives, or auto-refresh when I click into something relevantly related.

I don't see why DBeaver can't behave the same way, smartly for the user, without compromising security or performance.

If it times out or loses connection or whatever (maybe just memory is cached to storage, or virtual memory), DBeaver can think "Oh! The user is back. Is the connection still good? No? OK, refresh." All quietly and invisibly. Refresh of course doesn't necessarily mean "re-connect whenever the user clicks back to the app", it could mean "when the user wants to do something that requires a connection, check the connection, or refresh it, or when the connection is actually needed, check it and refresh it or prompt to reconnect if necessary".

See what I mean? Most apps I use handle this intelligently. I don't see why DBeaver can't. There's nothing special about my setup. I use MAMP Pro for local MySQL 5 hosting, and my only other connections are external and MySQL 5 and MySQL 8. However, the disconnect error happens no matter what. It just keeps happening. I can't see why, it doesn't matter which connection. I just think DBeaver is not smart with handling the connection for the user when they haven't used it recently. Even FileZilla makes it clear what happened and what to do, and it automatically handles a lot of things anyway. It's smart. I don't know what's going on with DBeaver.

**Comment by hoxbog** [comment]
I am also having this same issue with Dbeaver 23.1.4.202308061552 community / Windows 10 / Connecting to a remote MariaDB 10.6.12-MariaDB-cll-lve.  

- If I connect to the same DB with Heidi, there is no problem at all - it's rock solid.  Ditto Workbench.
- I also have a MariaDB 11.0.2-MariaDB instance on localhost which is rock solid via dbeaver.

This problem is so disruptive that I cannot work on the remote server with dbeaver which is a shame because I really like this tool.
[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/12340121/dbeaver-debug.log)

**Comment by athinboy** [comment]
same issue!
23.1.3.202307231901 + vpn+mysql
very frequently!

**Comment by AshwinSarath** [comment]
I am receiving a data Invalidated message, but I am able to query the data from database. 
This might be a typo error?

**Reply by Matvey16** [reply]
What do you mean by typo? This message means that your datasorce connections were checked and reconnected if needed. This happens after long idle time most of the time. You should be able to query data after invalidation

**Comment by rm-minus-r-star** [comment]
Same issue with DBeaver Community Ed Version 23.1.1.202306251800. I upgraded to Version 23.2.3.202310221533 and it's apparently working now.

**Reply by hoxbog** [reply]
I am on current versio n23.2.5.202311191730 and unfortunately still getting the the same behaviour.

**Reply by rm-minus-r-star** [reply]
My luck only lasted a few weeks also. I've actually had to stop using DBeaver. Now I just write/save queries in a text editor and paste into the psql shell. I can't find a client that works. It _might_ have something to do with the server -- the db that I have these issues with shuts down the compute after five minutes of inactivity to save on costs, whereas the Azure servers that we use at work are always on, and I don't have as many of these issues with DBeaver there. That said, DBeaver evidently doesn't handle this very well, as when this happens in the psql shell, I simply get 

> FATAL:  terminating connection due to administrator command
> SSL connection has been closed unexpectedly
> The connection to the server was lost. Attempting reset: 
> Succeeded.
> SSL connection (protocol: TLSv1.3, cipher: TLS_AES_256_GCM_SHA384, compression: off)
> DB=> 

and then I simply arrow up and hit enter to re-submit the query. It works every time. I'm guessing DBeaver trips over itself trying to do this.

**Comment by vcharette** [comment]
Have you looked under  `Edit Connection -> Connection settings -> Initialization -> Keep-Alive`, it seems like the default value may be `0` which disables the keep-alive functionality.

**Reply by hoxbog** [reply]
Hello, yes but changing this does not resolve the behaviour.  You can look at the errors in the logfile I attached to my previous post.

**Comment by Aigul9** [comment]
have you tried changing 
![image](https://github.com/dbeaver/dbeaver/assets/46625461/13b90c94-7846-42f3-81f7-5bba6c583b57)
in driver props?

**Comment by dolzenko** [comment]
NB. These can be just disabled through Preferences > User Interface > Notifications

**Comment by tispratik** [comment]
Nothing is helping. Neither the `socketTimeout` nor the `Keep-Alive`. I am on a Mac, Community Edition. Connection via SSL and DBeave is only allowing 1 query to run at a time, then I have to invalidate/revalidate the connection, run 1 query, repeat. Its very time consuming.
